addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -767,83 +801,128 @@ class EnvironmentMountService extends Service {
-    let studyInfo = [];
-    if (studyIds && studyIds.length) {
-      const [studyService, studyPermissionService] = await this.service(['studyService', 'studyPermissionService']);
-
-      studyInfo = await Promise.all(
-        _.map(studyIds, async studyId => {
-          try {
-            const { id, name, category, resources } = await studyService.mustFind(requestContext, studyId);
-            // Find out if the current user has Read/Write access
-            const uid = _.get(requestContext, 'principalIdentifier.uid');
-            const studyPermission = await studyPermissionService.findByUser(requestContext, uid);
-            const writeable = _.includes(studyPermission.readwriteAccess, studyId) || category === 'My Studies';
-            return { id, name, category, resources, writeable };
-          } catch (error) {
-            // Because the studies update periodically we cannot
-            // guarantee consistency so filter anything invalid here
-            return { name: '', resources: [] };
-          }
-        }),
-      );
-    }
-    return studyInfo;
-  }
-
-  async _validateStudyPermissions(requestContext, studyInfo) {
-    let permissions = {};
-    if (studyInfo.length) {
-      // Get requested study IDs
-      const requestedStudyIds = studyInfo.map(study => study.id);
-
-      // Retrieve and verify user's study permissions
-      const [studyPermissionService, studyService] = await this.service(['studyPermissionService', 'studyService']);
-      const storedPermissions = await studyPermissionService.getRequestorPermissions(requestContext);
-
-      // If there are no stored permissions, use an empty permissions object
-      permissions = storedPermissions || studyPermissionService.getEmptyUserPermissions();
+  // @private
+  // TODO - this is a temp solution, see my comments in the getStudyAccessInfo() method
+  async getStudies(requestContext, studyIds) {
+    if (_.isEmpty(studyIds)) return [];
+
+    const [studyService] = await this.service(['studyService']);
+    const uid = _.get(requestContext, 'principalIdentifier.uid');
+    const getInfo = async studyId => {
+      const studyEntity = await studyService.getStudyPermissions(requestContext, studyId);
+      if (!hasAccess(studyEntity, uid)) throw this.boom.forbidden(`You don't have access to study "${studyId}"`, true);
+      const { read, write } = accessLevels(studyEntity, uid);
+      const {
+        id,
+        category,
+        bucket,
+        region,
+        awsPartition,
+        folder,
+        resources = [],
+        // roleArn, this not available from the study entity, it needs to be checked out
+        kmsArn,
+      } = studyEntity;
+
+      return {
+        id,
+        category,
+        bucket,
+        region,
+        awsPartition,
+        prefix: folder,
+        resources,
+        // roleArn, this not available from the study entity, it needs to be checked out
+        kmsArn,
+        readable: read,
```

Comment:
Where is the `readable` parameter being used in the services?

Reply:
This is mainly done to be explicit, consistent and handle the accessType = 'writeonly', even though we don't have code that handles 'writeonly' all the way. It is better to be explicit in this case.  The purpose of this method is to tell us information about the mount, with this new code, we are explicitly list out the permissions.